ADVANCED.md

```diff
@@ -11,26 +11,50 @@
 # Headless Core
 
 - [Headless Core Concepts](#headless-core-concepts)
-- [Headless Core Application](#headless-core-application)
+- [Working with the Headless Core](#working-with-the-headless-core)
+- [Headless Core Reference](#headless-core-reference)
 
 ## Headless Core Concepts
 
-The core is a separate library which can be used for any JavaScript based implementation.
+```
+                                    |
+    @elastic/react-search-ui        |   @elastic/search-ui
+                                    |
+                                    |
+          SearchProvider <--------------- SearchDriver
+              |     |               |          |
+   State /    |     |               |          | State /
+   Actions    |     |               |          | Actions
+              |     |               |          |
+        Components  |               |          |
+              |     |               |          |
+              v     v               |          v
+------------------------------------+----------------------------
+              |     |                          |
+              v     v                          v
+          Using     Headless Usage       Headless Usage outside
+     Components     in React             of React
+```
 
-> [The Search UI Core](https://github.com/elastic/search-ui/tree/master/packages/search-ui).
+The core is a separate, vanilla JS library which can be used for any JavaScript based implementation.
 
-All of the components in this library use the headless core under the hood.
+> [@elastic/search-ui](https://github.com/elastic/search-ui/tree/master/packages/search-ui)
 
-You can work directly with the core if you need more than the Components offer.
+One way to think of the Headless Core is as a search experience without a view. Instead of providing UI like search boxes
+and dropdowns, it provides the underlying "state" and "actions" associated with that UI. For instance, instead of
+a "search box", the core has a `searchTerm` property which represents the value of the search box (the state), and a
+`setSearchTerm` method to update that searchTerm (the action).
 
-The Headless Core is managed by the `SearchProvider`, which provides a **"Context"**.
+Search UI lets you work directly with "state" and "actions", which means you are not restricted to using
```

Comment:
All of the components in this library use the Headless Core under the hood. For instance, Search UI provides `SearchBox` component for collecting input from a user. But you are not restricted to using just that component. Since Search UI lets you work directly with "state" and "actions", you could use any type of input you want! As long as your input or component calls the Headless Core's `setSearchTerm` action, it will "just work". This gives you maximum flexibility over your experience if you need more than the components in Search UI have to offer.

Reply:
Used it with just a few small tweaks.